CDCgov/prime-reportstream
# Adjust convert/translate schemas to handle ELIMS test message 40 (HL7->FHIR->HL7)

This PR involves a variety of updates to convert/translate schemas in support of lossless conversion of a sample rhapsody message from HL7v2 to FHIR and back to HL7v2. Note that the diff for some files may be much easier to view in an IDE than in github.

Test Steps:
1. Update the dev-elims receiver in `organizations.yml` to be active and reload the settings.
2. Send a report to ReportStream using message 40 in [this folder](https://cdc.sharepoint.com/teams/ReportStream/Shared%20Documents/Forms/AllItems.aspx?RootFolder=%2Fteams%2FReportStream%2FShared%20Documents%2FELIMS%2DELR) as the message body.
3. Compare the resulting HL7 to the input HL7 and observe that any differences are appropriately documented in the [notes spreadsheet](https://cdc.sharepoint.com/:x:/r/teams/ReportStream/Shared%20Documents/ELIMS-ELR/[40_Message_RS_processing_notes.xlsx](https://cdc.sharepoint.com/:x:/r/teams/ReportStream/Shared%20Documents/ELIMS-ELR/40_Message_RS_processing_notes.xlsx?d=w0e69763787ea46ffa7187cee9417b41d&csf=1&web=1&e=8bpFIa)?d=w0e69763787ea46ffa7187cee9417b41d&csf=1&web=1&e=8bpFIa)

## Changes
- Added custom extension for MSH.15 and MSH.16
- Removed CE.6 from the `coding_1.version` in `CodeableConcept.yml` (seemed incorrect and could not find justification for this)
- Add FHIR mapping using custom extension for MSH.21
- Add CLIA universal id in `MessageSender.yml`
- Use `CodeableConcept` instead of `Reference/Organization` when mapping OBX.15 to FHIR (FHIR->HL7v2 can still handle either, this was done because Organization is incomplete in getting all 6 fields from OBX.15)
- Fix reading of `CodeableConcept` to populate OBX.15
- Add FHIR->HL7v2 translation of OBX.16
- Add condition to only populate OBX.25 if the practitioner has a MDIR code
- Fix bad constant in `ce-coded-with-exceptions` to allow CWE.9 to be properly populated
- Add conditions for PID.10.3, CX.4, and CX.6 to avoid unnecessarily populating
- Update XAD mapping to more closely match spec
- Add condition for XTN.12 to not use for email
- Add placeholder for sender transform for Rhapsody and override default transform for SFT.1.1
- Removed default value for SFT.1.2 (seemed incorrect and could not find justification for this)
- Fix bug with FHIRRouter to only consider org filters that match the current topic (and associated test update)
- Updates to many integration tests (see update to readme for explanation of some new files)

## Checklist

### Testing
- [ ] Added tests? (Skipped this for now since the message is still not fully end-to-end lossless, will add integration tests in a later PR)

## Linked Issues
- Addresses #10027

## To Be Done
- The remaining differences in the notes spreadsheet should continue to be resolved, but perhaps as a separate ticket given the overall complexity.
